libbeat/monitoring/report/report.go

```diff
@@ -128,7 +129,9 @@ func collectSubObject(cfg *conf.C) *conf.C {
 	for _, field := range cfg.GetFields() {
 		if obj, err := cfg.Child(field, -1); err == nil {
 			// on error field is no object, but primitive value -> ignore
-			out.SetChild(field, -1, obj)
+			if err := out.SetChild(field, -1, obj); err != nil {
+				fmt.Fprintf(os.Stdout, "Failed to set the sub-configuration: %v\n", err)
```

Comment:
I don't think we should write it to the output.

I think it's better to create an exception for the linter by adding `//nolint:errcheck // this error is safe to ignore`.